### PR TITLE
Added custom contract resolvers to handle STJ property name attributes

### LINKED ISSUE #113: Deserialization of restorationjob's patch.json fails to deserialize namespace value
**Describe the bug**
The restorationjob fails to deserialize the namespace value contained within the patch.json file causing a null namespace to be passed to `ReadNamespacedDeploymentWithHttpMessagesAsync` which throws an ArgumentNullException.

**Mention the platform you are using**
Windows 11, VS2022, Kubernetes KinD cluster v1.23.5

**To Reproduce**
Install VS2022 B2K extension, F5 a project, connect to cluster.
Restorationjob pod starts, exceptions and CrashLoopBackoffs.
404 always returned from previous working browser url.

**Expected behavior**
Should not exception, but no idea beyond that as not got any further.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**
Here's the pod log from restoration job:
```
at k8s.AbstractKubernetes.ReadNamespacedDeploymentWithHttpMessagesAsync(String name, String namespaceParameter, Nullable`1 pretty, IReadOnlyDictionary`2 customHeaders, CancellationToken cancellationToken)

at k8s.KubernetesExtensions.ReadNamespacedDeploymentAsync(IBasicKubernetes operations, String name, String namespaceParameter, Nullable`1 pretty, CancellationToken cancellationToken)

at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass63_0.<<GetV1DeploymentAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 683

--- End of stack trace from previous location where exception was thrown ---

at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass75_0`1.<<ClientInvokeWrapperAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 1058

--- End of stack trace from previous location where exception was thrown ---

at Microsoft.BridgeToKubernetes.Common.Utilities.WebUtilities.RetryUntilTimeAsync(Func`2 action, TimeSpan maxWaitTime, CancellationToken cancellationToken) in /src/common/Utilities/WebUtilities.cs:line 112

at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.ClientInvokeWrapperAsync[T](Func`1 handler, String operation, CancellationToken cancellationToken) in /src/common/Kubernetes/KubernetesClient.cs:line 1054

at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass24_0.<<ListPodsForDeploymentAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 202

--- End of stack trace from previous location where exception was thrown ---

at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.<>c__DisplayClass75_0`1.<<ClientInvokeWrapperAsync>b__0>d.MoveNext() in /src/common/Kubernetes/KubernetesClient.cs:line 1058

--- End of stack trace from previous location where exception was thrown ---

at Microsoft.BridgeToKubernetes.Common.Utilities.WebUtilities.RetryUntilTimeAsync(Func`2 action, TimeSpan maxWaitTime, CancellationToken cancellationToken) in /src/common/Utilities/WebUtilities.cs:line 112

at Microsoft.BridgeToKubernetes.Common.Kubernetes.KubernetesClient.ClientInvokeWrapperAsync[T](Func`1 handler, String operation, CancellationToken cancellationToken) in /src/common/Kubernetes/KubernetesClient.cs:line 1054

at Microsoft.BridgeToKubernetes.DevHostAgent.RestorationJob.RestorationJobApp._GetAgentEndpointAsync(DeploymentPatch deploymentPatch, CancellationToken cancellationToken) in /src/devhostagent.restorationjob/RestorationJobApp.cs:line 236","RemoteStackTraceString

```

I wrote a small unit test to confirm suspicions around using Newtonsoft.Json rather than System.Text.Json.  This just uses the current DeploymentPatch.json file in the devhostAgent.restorationjob.tests:
```
        [Fact]
        public void EnsureCanDeserializePatch()
        {
            string patchStateJson = File.ReadAllText(Path.Combine("TestData", "DeploymentPatch.json"));
            var deploymentPatch = JsonHelpers.DeserializeObject<DeploymentPatch>(patchStateJson);

            string name = deploymentPatch.Deployment.Name();
            string ns = deploymentPatch.Deployment.Namespace();

            Assert.Equal("bikes", name);
            Assert.Equal("dev", ns);
        }
```

The test fails as the ns variable is null.  The reason being the generated k8s.Models.V1ObjectMeta type's NamespaceProperty only supports a System.Text.Json.JsonPropertyName attribute so Newtonsoft.Json will not know to deserialize from the "namespace" name.  There are many other type properties with this attribute but their name only differs by case whereas this one differs by more than case alone.

### PR DESCRIPTION
**Pull request by markphillips100**
This PR fixes #113 by adding 2 new Json.Net contract resolvers to replace usage of existing `DefaultContractResolver` and `CamelCasePropertyNamesContractResolver` implementations.  Their primary purpose being to retain the existing camel casing behaviour or not as required, with support for deserializing property names where the property is using a `System.Text.Json.Serialization.JsonPropertyNameAttribute`.

Some notes:
1. Added some unit tests to validate the behaviour.
2. Did not change the base type of `PropertyIgnoreSerializerContractResolver` as was unsure of the impact.
3. Modified all ContractResolver assignments to use the appropriate new resolvers.
4. There appears to be a broken unit test unrelated to this change, notably `ValidateDateTimeFormatForDefaultDateTimeString()`.
5. Added an msbuild property to build\AssemblyVersion.proj to allow me to not include my username in version.  My local username ends in an underscore (no idea why) and this caused a problem building.

### CODE REVIEW

**Comment by markphillips100** [comment]
@microsoft-github-policy-service agree [company="default"]

**Comment by markphillips100** [comment]
@microsoft-github-policy-service agree

**Review by hsubramanianaks — COMMENTED** [review]
@markphillips100 Thank you for the PR. I appreciate it. 
ValidateDateTimeFormatForDefaultDateTimeString - this UT is not failing in our CI/CD pipelines, and I manually tested them in my local which doesn't fail either. It Should be something related to your local machine.
we will review your changes further and let you know. 

Thanks,
Hari

**Comment by markphillips100** [comment]
Is someone else going to merge the latest changes?  Also, any chance this can be reviewed and released soon?

File: build/AssemblyVersion.proj
Comment (elenavillamil) [inline build/AssemblyVersion.proj:6]: Can you keep this change locally only?

File: src/routingmanager/Properties/launchSettings.json
Comment (elenavillamil) [inline src/routingmanager/Properties/launchSettings.json:1]: Does this file need to be checked in or can it be kept locally? Is it needed for UTs to run?

**Review by elenavillamil — COMMENTED** [review]
Hi! Thank you very much for your PR. I have left some small comments. The main changes look good!

File: build/AssemblyVersion.proj
Comment (Eneuman) [reply build/AssemblyVersion.proj:6]: I think this would be a great addition to the code. I also have the same problem with the user name and can't build the code so I will not be able to contribute as I would like to do. Just my 2 cents.

Comment (elenavillamil) [reply build/AssemblyVersion.proj:6]: Sounds good. We can keep it.

**Comment by elenavillamil** [comment]
> Is someone else going to merge the latest changes? Also, any chance this can be reviewed and released soon?

Hi @markphillips100, sorry for the delay we were focused on the switch to dotnet6. We have had some time in the past couple days to look at this PR and once we get green light from our testing team Monday we should be able to merge it. If you can remove the launchSettings.json file it would be ideal, but it is not a big issue if it stays.

File: src/routingmanager/Properties/launchSettings.json
Comment (markphillips100) [reply src/routingmanager/Properties/launchSettings.json:1]: It is not needed.  Visual Studio added for some reason and I missed it.

Sorry been very busy, is it possible someone else can update the PR?

File: build/setup-kubectl.sh
Comment (markphillips100) [reply build/setup-kubectl.sh:1]: Prior to the PR, when I attempted to build an image for restorationjob it would fail when executing the last RUN command that runs the 2 sh files.  I concluded that bash was not installed in the base image so just changed the two script's to use sh instead.  No idea what "ash" is either?

You are welcome to change it back to bash, or remove my change - I'm really busy at the moment.  I would expect to be able to build the image locally though, yes/no?

Is bash supposed to be installed?  Just not sure how the image ever built on the commit I pulled is all.

Comment (Tatsinnit) [reply build/setup-kubectl.sh:1]: 💡cool, “ash” is default shell for alpine (ash is minimalist shell and sometimes keeps security by design) : https://wiki.alpinelinux.org/wiki/Change_default_shell

Since it is unintentional let’s revert it, I have few things to do but we can collaborate on this and I can remove this line sompoint today.

I can relive this in couple of hours, might add PR to your fork!!

Comment (markphillips100) [reply build/setup-kubectl.sh:1]: Cool, thanks for boosting my knowledge re "ash".  Seems like I would have been better changing both to "ash" is my guess.  Oh well never mind.

Thanks for understanding re availability.

Comment (Tatsinnit) [reply build/setup-kubectl.sh]: 👍 cool, Fyi @markphillips100 - I have removed the other file discussed in one of comment and also reverted this change to whatever was there in main. :) please take a look and see if anything else got checked in which is not needed?

Comment (markphillips100) [reply build/setup-kubectl.sh]: Looks good @Tatsinnit .  As for `launchSettings.json`, that is not needed.  My bad.

Comment (Tatsinnit) [reply build/setup-kubectl.sh]: All good @markphillips100 - done, 👍 - please take a look and I've removed the other local launch setting file as well. Thanks.

File: src/common.json/JsonHelpers.cs
Comment (Tatsinnit) [inline src/common.json/JsonHelpers.cs:125]: 💡☕️ Thanks @markphillips100 for this, I will fyi @hsubramanianaks and @elenavillamil for there eyes around his question and `todo` - so that folks have noticed this question from you.

Comment (markphillips100) [reply src/common.json/JsonHelpers.cs:125]: Thanks.  I didn't want to blanket apply the new resolver without knowing for sure the impact.

File: build/setup-kubectl.sh
Comment (markphillips100) [reply build/setup-kubectl.sh]: Nope.  no need for any of the launchSettings.json.  Visual Studio added them for me for some reason and I failed to notice.

Comment (markphillips100) [reply build/setup-kubectl.sh]: I confirm current changes building okay

Comment (Tatsinnit) [reply build/setup-kubectl.sh]: Awesome, done, I have removed and pushed this file as well `src/devhostagent/Properties/launchSettings.json` ! thanks will it be cool if you check that all looks sane at your end whenever you have time. Thanks so much for this.

Comment (markphillips100) [reply build/setup-kubectl.sh]: Yeah I just got out of meetings (thankfully) so was able to check a few minutes ago, and can confirm your latest builds and tests good also.

File: src/common.json/JsonHelpers.cs
Comment (elenavillamil) [reply src/common.json/JsonHelpers.cs:125]: Thanks @Tatsinnit ! Ya I did noticed, I think leaving it as a TODO seems the right step for this PR. Changing this would be larger change with larger impact and since we recently took over this project from a different team I am not sure if there is any historical reasoning behind this.

File: build/setup-kubectl.sh
Comment (elenavillamil) [reply build/setup-kubectl.sh]: Hi! Sorry this is still opened, I just wanted to give you an update. Our testing team found one scenario that did not work. In particular, restoration job was having issues. We are working on it before merging the change.

Comment (markphillips100) [reply build/setup-kubectl.sh]: Anything I can do?

Comment (elenavillamil) [reply build/setup-kubectl.sh]: You are all good. I worked on this yesterday and fix the issue (it was on our end - docker images on dev environment had been overridden). Testing team gave green light for this change now.

**Review by elenavillamil — APPROVED** [review]

**Pull request merged**
